radix-engine/src/blueprints/intent_hash_store/package.rs

```diff
@@ -0,0 +1,280 @@
+                ),
+                export: INTENT_HASH_STORE_CREATE_EXPORT_NAME.to_string(),
+            },
+        );
+
+        let schema = generate_full_schema(aggregator);
+        let blueprints = btreemap!(
+            INTENT_HASH_STORE_BLUEPRINT.to_string() => BlueprintDefinitionInit {
+                outer_blueprint: None,
+                dependencies: btreeset!(
+                ),
+                feature_set: btreeset!(),
+                schema: BlueprintSchemaInit {
+                    generics: vec![],
+                    schema,
+                    state: BlueprintStateSchemaInit {
+                        fields,
+                        collections,
+                    },
+                    events: BlueprintEventSchemaInit::default(),
+                    functions: BlueprintFunctionsSchemaInit {
+                        virtual_lazy_load_functions: btreemap!(),
+                        functions,
+                    },
+                },
+
+                royalty_config: RoyaltyConfig::default(),
+                auth_config: AuthConfig {
+                    function_auth: btreemap!(
+                    ),
+                    method_auth: MethodAuthTemplate::Static {
+                        auth: btreemap!(),
+                        outer_auth: btreemap!(),
+                    },
+                },
+            }
+        );
+
+        PackageDefinition { blueprints }
+    }
+
+    #[trace_resources(log=export_name)]
+    pub fn invoke_export<Y>(
+        export_name: &str,
+        input: &IndexedScryptoValue,
+        api: &mut Y,
+    ) -> Result<IndexedScryptoValue, RuntimeError>
+    where
+        Y: ClientApi<RuntimeError>,
+    {
+        match export_name {
+            INTENT_HASH_STORE_CREATE_EXPORT_NAME => {
+                api.consume_cost_units(FIXED_LOW_FEE, ClientCostingReason::RunNative)?;
+
+                let input: IntentHashStoreCreateInput = input.as_typed().map_err(|e| {
+                    RuntimeError::ApplicationError(ApplicationError::InputDecodeError(e))
+                })?;
+
+                let rtn = IntentHashStoreBlueprint::create(input.address_reservation, api)?;
+
+                Ok(IndexedScryptoValue::from_typed(&rtn))
+            }
+
+            _ => Err(RuntimeError::ApplicationError(
+                ApplicationError::ExportDoesNotExist(export_name.to_string()),
+            )),
+        }
+    }
+}
+
+#[derive(Debug, Clone, ScryptoSbor)]
+pub enum IntentHashStatus {
+    CommittedSuccess,
+    CommittedFailure,
+    Cancelled,
+}
+
+#[derive(Debug, Clone, ScryptoSbor)]
+pub struct IntentHashStoreSubstate {
+    pub start_epoch: u64,
+    pub start_partition: u8,
+
+    // parameters
+    pub partition_range_start: u8,
+    pub partition_range_end: u8,
+    pub epochs_per_partition: u64,
+}
+
+impl IntentHashStoreSubstate {
+    pub fn partition_of(&self, epoch: u64) -> Option<u8> {
```

Comment:
Can we add a rust doc explaining what this does?

I'd prefer call this `partition_for_expiry_epoch`

Reply:
Maybe it would be clearer if it was a custom enum rather than option? EG:
```
pub enum ResolvedTransactionExpiryEpochPartition {
    Partition(u8),
    EpochNotInCurrentTrackableRange(Epoch),
}
```

Reply:
Happy to rename, but don't feel necessity to add further abstraction.

Will just add some comments to the interface.

Reply:
It should also take an `Epoch` rather than a u64 :)